pkg/option/config.go

```diff
@@ -3113,8 +3114,23 @@ func InitConfig(programName, configName string) func() {
 			} else {
 				// replace deprecated fields with new fields
 				ReplaceDeprecatedFields(m)
-				err := MergeConfig(m)
-				if err != nil {
+
+				// validate the config-map
+				var msgError = "Incorrect config-map flag"
+				for key, value := range m {
+					if val := fmt.Sprintf("%v", value); val != "" {
+						flags := cmd.Flags()
+						// check whether the flag exists
+						if flag := flags.Lookup(key); flag != nil {
+							// validate the value of flag
+							if err := flag.Value.Set(val); err != nil {
+								log.WithError(err).Fatal(msgError + "  " + key)
```

Comment:
Perhaps this just needs to be a warning?

Reply:
> Perhaps this just needs to be a warning?

No. The ConfigMap validation should be same as for CLI.